src/components/recipedetails/recipeDetails.jsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react'
+import { useLocation } from "react-router-dom";
+import "./recipeDetails.css"
+
+// this page needs to be designed
+
+function RecipeDetails() {
+  const location = useLocation();
+  const { recipeData } = location.state;
```

Comment:
to get an array of the ingredients:
const ingredients = recipeData.sections[0].components.map(ingredient => ingredient.raw_text);

Then we can map this to display in your ingredients box